gazebo_ros2_control/include/gazebo_ros2_control/gazebo_system.hpp

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef GAZEBO_ROS2_CONTROL__GAZEBO_SYSTEM_HPP_
+#define GAZEBO_ROS2_CONTROL__GAZEBO_SYSTEM_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "angles/angles.h"
+
+#include "control_toolbox/pid_ros.hpp"
+
+#include "gazebo_ros2_control/gazebo_system_interface.hpp"
+
+#include "std_msgs/msg/bool.hpp"
+
+namespace gazebo_ros2_control
+{
+
+class GazeboSystem : public GazeboSystemInterface
+{
+public:
+  // Documentation Inherited
+  hardware_interface::return_type configure(const hardware_interface::HardwareInfo & system_info)
+  override;
+
+  // Documentation Inherited
+  std::vector<hardware_interface::StateInterface> export_state_interfaces() override;
+
+  // Documentation Inherited
+  std::vector<hardware_interface::CommandInterface> export_command_interfaces() override;
+
+  // Documentation Inherited
+  hardware_interface::return_type start() override;
+
+  // Documentation Inherited
+  hardware_interface::return_type stop() override;
+
+  // Documentation Inherited
+  hardware_interface::return_type read() override;
+
+  // Documentation Inherited
+  hardware_interface::return_type write() override;
+
+  // Documentation Inherited
+  bool initSim(
+    rclcpp::Node::SharedPtr & model_nh,
+    gazebo::physics::ModelPtr parent_model,
+    const urdf::Model * const urdf_model,
+    std::vector<transmission_interface::TransmissionInfo> transmissions,
+    sdf::ElementPtr sdf) override;
+
+  /// \brief callback to get the e_stop signal from the ROS 2 topic
+  void eStopCB(const std::shared_ptr<std_msgs::msg::Bool> e_stop_active);
+
+private:
```

Comment:
This header is installed. How about using the PIMPL pattern to hide these private members, so that the class can be changed without breaking ABI?